runtimes/rust/tests/runtime.rs

```diff
@@ -0,0 +1,1249 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[macro_use]
+extern crate pretty_assertions;
+
+use exonum::{
+    blockchain::{
+        config::{GenesisConfig, GenesisConfigBuilder, InstanceInitParams},
+        Blockchain, BlockchainBuilder, BlockchainMut, Schema as CoreSchema,
+    },
+    helpers::{generate_testnet_config, Height, ValidatorId},
+    messages::{AnyTx, Verified},
+    runtime::{
+        CallInfo, ExecutionContext, InstanceSpec, InstanceStatus, Mailbox, Runtime,
+        WellKnownRuntime,
+    },
+};
+use exonum_crypto::Hash;
+use exonum_derive::exonum_interface;
+use exonum_derive::*;
+use exonum_merkledb::{access::AccessExt, BinaryValue, ObjectHash, Patch, Snapshot, SystemSchema};
+use exonum_rust_runtime::{
+    ArtifactId, CallContext, Caller, DefaultInstance, DispatcherError, ErrorMatch, ExecutionError,
+    InstanceId, RustRuntime, Service, ServiceFactory, SnapshotExt, SUPERVISOR_INSTANCE_ID,
+};
+use futures::{sync::mpsc, Future};
+use serde_derive::*;
+
+use std::{
+    collections::BTreeMap,
+    sync::{Arc, Mutex},
+};
+
+macro_rules! impl_binary_value_for_bincode {
```

Comment:
I wonder whether it makes sense to put this macro in the MerkleDB crate. `bincode` serialization is similarly used in several other tests (e.g., in #1657).

Reply:
I think the best way is to add `bincode` support to the `BinaryValue` derive.

Reply:
Yeah, makes sense.